src/services/completions.ts

```diff
@@ -186,14 +185,24 @@ namespace ts.Completions {
             kindModifiers: SymbolDisplay.getSymbolModifiers(symbol),
             sortText: "0",
             source: getSourceFromOrigin(origin),
-            // TODO: GH#20619 Use configured quote style
-            insertText: needsConvertPropertyAccess ? `["${name}"]` : undefined,
-            replacementSpan: needsConvertPropertyAccess
-                ? createTextSpanFromBounds(findChildOfKind(propertyAccessToConvert, SyntaxKind.DotToken, sourceFile)!.getStart(sourceFile), propertyAccessToConvert.name.end)
-                : undefined,
-            hasAction: trueOrUndefined(needsConvertPropertyAccess || origin !== undefined),
+            hasAction: trueOrUndefined(origin !== undefined),
             isRecommended: trueOrUndefined(isRecommendedCompletionMatch(symbol, recommendedCompletion, typeChecker)),
+            ...getInsertTextAndReplacementSpan(),
         };
+
+        function getInsertTextAndReplacementSpan(): { insertText?: string, replacementSpan?: TextSpan } {
+            if (kind === CompletionKind.Global) {
+                if (typeChecker.isMemberSymbol(symbol)) {

```

Comment:
seems kinda weird that we add the symbol, then later on ask the checker if they were the ones we added.. i wounder if we can do something different here..

Reply:
alas.. i could not come up with a better idea here myself. ignore my previous rant.

Reply:
We could use `SymbolOriginInfoMap`, would that be a good idea?

Reply:
yeah.. that is not bad..